components/common/locale.ts

```diff
@@ -40,30 +40,53 @@ interface SplitStringForTagResult {
  * @param {object} replacements - replacements for specific translation, replacement should be defined as {{key}}
  * @returns {SplitStringForTagResult}
  */
-export const getLocaleTags = (key: string, replacements?: Record<string, string>) => {
+export const getLocaleWithTag = (key: string, replacements?: Record<string, string>) => {
   const text = getLocale(key, replacements)
-  return splitStringForTag(text, '$1', '$2')
+  return splitStringForTag(text)
+}
+
+/**
+ * Returns text for translations with a multiple HTML tags
+ * (like a link or button)
+ * @param {string} key - translation identifier
+ * @param {number} tags - how many tags is in translation
+ * @param {object} replacements - replacements for specific translation, replacement should be defined as {{key}}
+ * @returns {SplitStringForTagResult}
+ */
+export const getLocaleWithTags = (key: string, tags: number, replacements?: Record<string, string>) => {
+  let text = getLocale(key, replacements)
+  let result = []
+  for (let i = 1; i <= tags; i++) {
+    const split = splitStringForTag(text, (i * 2 - 1))
+    text = split.afterTag || ''
+    if (i !== tags) {
+      split.afterTag = ''
+    }
+    result.push(split)
+  }
+  return result
 }
 
 /**
  * Allows an html or xml tag to be injected in to a string by extracting
  * the components of the string before, during and after the tag.
  * Usage:
- *    splitStringForTag('my string with some $1bold text$2', '$1', '$2')
+ *    splitStringForTag('my string with some $1bold text$2')
  *
  * @export
  * @param {string} text
- * @param {string} tagOpenPlaceholder
- * @param {string} tagClosePlaceholder
+ * @param {number} tagStartNumber - starting number for the tag
  * @returns {SplitStringForTagResult}
  */
-export function splitStringForTag (text: string, tagOpenPlaceholder: string, tagClosePlaceholder: string): SplitStringForTagResult {
+export function splitStringForTag (text: string, tagStartNumber: number = 1): SplitStringForTagResult {
```

Comment:
The number argument seems a little strange to use, whereas the previous signature was very explicit. Maybe leave this function signature as-is and build the `$n` strings in `getLocaleWithTags`?

Reply:
this just simplifies the signature as in most cases people would just do `splitStringForTag(text)` and move all complexity in `splitStringForTag`